// ngx-scrollbar/esm2022/lib/utils/ng-attr.directive.mjs
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class NgAttr {
    constructor(el) {
        this.el = el;
    }
    set ngAttr(attrs) {
        for (const [key, value] of Object.entries(attrs)) {
            this.el.nativeElement.setAttribute(key, value);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.5", ngImport: i0, type: NgAttr, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.5", type: NgAttr, isStandalone: true, selector: "[ngAttr]", inputs: { ngAttr: "ngAttr" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.5", ngImport: i0, type: NgAttr, decorators: [{
            type: Directive,
            args: [{
                    selector: '[ngAttr]',
                    standalone: true
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { ngAttr: [{
                type: Input
            }] } });
