// ngx-lottie/esm2020/lib/cacheable-animation-loader/lottie-cache.module.mjs
import { NgModule } from '@angular/core';
import { AnimationLoader } from '../animation-loader';
import { CacheableAnimationLoader } from './cacheable-animation-loader';
import * as i0 from "@angular/core";
export class LottieCacheModule {
    static forRoot() {
        return {
            ngModule: LottieCacheModule,
            providers: [
                {
                    provide: AnimationLoader,
                    useExisting: CacheableAnimationLoader,
                },
            ],
        };
    }
}
/** @nocollapse */ LottieCacheModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.0", ngImport: i0, type: LottieCacheModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LottieCacheModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.0", ngImport: i0, type: LottieCacheModule });
/** @nocollapse */ LottieCacheModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.0", ngImport: i0, type: LottieCacheModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.0", ngImport: i0, type: LottieCacheModule, decorators: [{
            type: NgModule
        }] });
