// ngx-lottie/esm2020/server/lottie-server.module.mjs
import { NgModule, InjectionToken, APP_INITIALIZER } from '@angular/core';
import { TransferState } from '@angular/platform-browser';
import { appInitializerFactory } from './internals';
import * as i0 from "@angular/core";
export const LOTTIE_SERVER_OPTIONS = new InjectionToken('LottieServerOptions');
export class LottieServerModule {
    static forRoot(options) {
        return {
            ngModule: LottieServerModule,
            providers: [
                {
                    provide: LOTTIE_SERVER_OPTIONS,
                    useValue: options,
                },
                {
                    provide: APP_INITIALIZER,
                    useFactory: appInitializerFactory,
                    multi: true,
                    deps: [LOTTIE_SERVER_OPTIONS, TransferState],
                },
            ],
        };
    }
}
/** @nocollapse */ LottieServerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.0", ngImport: i0, type: LottieServerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
/** @nocollapse */ LottieServerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.0.0", ngImport: i0, type: LottieServerModule });
/** @nocollapse */ LottieServerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.0.0", ngImport: i0, type: LottieServerModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.0", ngImport: i0, type: LottieServerModule, decorators: [{
            type: NgModule
        }] });
